lib/core/src/Cardano/Wallet/DB/Sqlite.hs

```diff
@@ -103,51 +158,219 @@ TxIn
+                                    Checkpoints
+        -----------------------------------------------------------------------}
+
+        , putCheckpoint = \(PrimaryKey _wid) _cp -> error "unimplemented"
+
+        , readCheckpoint = \(PrimaryKey _wid) -> error "unimplemented"
+
+        {-----------------------------------------------------------------------
+                                   Wallet Metadata
+        -----------------------------------------------------------------------}
+
+        , putWalletMeta = \(PrimaryKey wid) meta ->
+            ExceptT $ runQuery conn $
+            selectWallet wid >>= \case
+                Just _ -> do
+                    updateWhere [WalTableId ==. wid]
+                        (mkWalletMetadataUpdate meta)
+                    pure $ Right ()
+                Nothing -> pure $ Left $ ErrNoSuchWallet wid
+
+        , readWalletMeta = \(PrimaryKey wid) ->
+            runQuery conn $
+            fmap (metadataFromEntity . entityVal) <$>
+            selectFirst [WalTableId ==. wid] []
+
+        {-----------------------------------------------------------------------
+                                     Tx History
+        -----------------------------------------------------------------------}
+
+        , putTxHistory = \(PrimaryKey _wid) _txs -> error "unimplemented"
+
+        , readTxHistory = \(PrimaryKey _wid) -> error "unimplemented"
+
+        {-----------------------------------------------------------------------
+                                       Keystore
+        -----------------------------------------------------------------------}
+
+        , putPrivateKey = \(PrimaryKey _wid) _key -> error "unimplemented"
+
+        , readPrivateKey = \(PrimaryKey _wid) -> error "unimplemented"
+
+        {-----------------------------------------------------------------------
+                                       Lock
+        -----------------------------------------------------------------------}
+
+        , withLock = \_action -> error "withLock to be implemented"
+
+        }
+
+----------------------------------------------------------------------------
+-- Conversion between Persistent table types and wallet types
+
+delegationToText :: W.WalletDelegation W.PoolId -> Maybe Text
+delegationToText W.NotDelegating = Nothing
+delegationToText (W.Delegating pool) = Just (W.getPoolId pool)
+
+delegationFromText :: Maybe Text -> W.WalletDelegation W.PoolId
+delegationFromText Nothing = W.NotDelegating
+delegationFromText (Just pool) = W.Delegating (W.PoolId pool)
+
+mkWalletEntity :: W.WalletId -> W.WalletMetadata -> Wallet
+mkWalletEntity wid meta = Wallet
+    { walTableId = wid
+    , walTableName = meta ^. #name . coerce
+    , walTablePassphraseLastUpdatedAt =
+        W.lastUpdatedAt <$> meta ^. #passphraseInfo
+    , walTableStatus = meta ^. #status
+    , walTableDelegation = delegationToText $ meta ^. #delegation
+    , walTableAddressScheme = Sequential
```

Comment:
This probably has to be addressed in another PR. Detection whether we are Random or Sequential

Reply:
Yes there is not yet reading/writing of the wallet state.
However we may even remove this field because the address scheme is fixed by a type parameter of DBLayer.

Reply:
exactly. This is in line what @KtorZ wrote on slack yesterday:
```
Then, we'll have one

- external seq wallet layer + seq db layer
- fully owned  seq wallet + seq db layer
- fully owned rnd wallet + rnd db layer
```
Address scheme is set for each db instance. Hence, `walTableAddressScheme` is probably superflouous

Reply:
removed address scheme